docs/detections/rules-ui-create.asciidoc

```diff
@@ -263,33 +263,37 @@ For example, if a rule has an interval of 5 minutes, no additional look-back tim
 [[create-esql-rule]]
 === Create an ES|QL rule
 
-IMPORTANT: This is a placeholder for future documentation. The following content is incomplete.
+preview::[]
 
 . Go to *Rules* -> *Detection rules (SIEM)* -> *Create new rule*. The *Create new rule* page displays.
-. To create a rule that uses ES|QL, select **ES|QL**, 
-then write a query. There are two types of ES|QL queries:
-
-.. **Aggregating query**: This is a query that uses the `STATS...BY` grouping commands. Query results cannot be matched with a particular document in Elasticsearch. For example:
+. Select **ES|QL**. 
+. Write an aggregating or non-aggregating type query:
 +
-[esql]
------
+TIP: Click the **ES|QL reference** button [book icon] for more information on ES|QL syntax. 
++
+** **Aggregating query**: Uses {ref}/esql-functions-operators.html#esql-agg-functions[`STATS...BY`] functions to aggregate alert data. An alert is generated for each segment of the query, but is not associated with a specific source event since aggregating queries process data from multiple alert documents. As a result, alerts from aggregating queries do not contain source event data.
++
+Example aggregating query: 
++
+[source,esql]
+----
 FROM logs*
 | STATS count = COUNT(host.name) BY host.name
 | SORT host.name
------
+----
 
-.. **Non-aggregating query**: This is a query that _does not_ use the `STATS...BY` grouping commands. Each row in the query results can be tracked to a source document in Elasticsearch. 
+** **Non-aggregating query**: Doesn't use `STATS...BY` functions and doesn't aggregate alert data. An alert is generated for each segment of the query, and each alert is associated with a source event. As a result, alerts from non-aggregating queries contain source event data.
```

Comment:
> functions and doesn't aggregate alert data

```suggestion
** **Non-aggregating query**: Doesn't use `STATS...BY` functions and doesn't aggregate source data. An alert is generated for each row in the result, and each alert can be linked to a source event through [metadata] operator. As a result, alerts from non-aggregating queries contain source event data merged with fields returned by query request
```

Reply:
row, column, table - something described in our tech docs: https://esql.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/esql.html
So, we can use it our advantage and add link to it in our doc.
And we can put in common area of ES|QL section: each row become an alert. That is cornerstone for how this rule type works

Reply:
Would you mind elaborating on the last part of your comment? I'm not clear on whether you're suggesting that we explain how the ES|QL rule generates alerts in these docs, or in the [core ES|QL docs](https://www.elastic.co/guide/en/elasticsearch/reference/master/esql.html).

Reply:
@vitaliidm in addition to explaining how ES|QL alerts are generated, do you think we need to mention any best practices for writing ES|QL rule queries? For example, if users drop certain fields in the query, will those fields be omitted from the alert document? Also, should they include certain fields to ensure they can use specific functionality in the alert details flyout, like the [Entities](https://www.elastic.co/guide/en/security/master/view-alert-details.html#entities-overview) overview?

Reply:
Oh yeah, I'm not suggesting that we go through all of our features to find fields that users need to keep or return. I'm more so asking if we need to highlight detail because I feel like it's pretty obscured and can affect users's ability to use certain functionality in the flyout or elsewhere.

Reply:
It's difficult to answer, it would depend which flyout(or smth else) functionality we refer. 

As I mentioned  I think it is sufficient enough to point, source document will be merged to alert for non-aggregatin query. Depends on it, user would be able to use other Security Solution features.

I think scope of features too big to mention each single one